### PR TITLE
[Kotlin Migration] SchedTest

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Kotlin Migration of SchedTest

### CODE REVIEW

**Comment by dae** [comment]
This looks like it's not based on latest main, as some of these changes were already made in https://github.com/ankidroid/Anki-Android/pull/11802

File: AnkiDroid/src/test/java/com/ichi2/libanki/sched/SchedTest.kt [hunk 1]
```diff
@@ -13,244 +13,238 @@
...
+import org.hamcrest.Matchers.nullValue
+import org.junit.Assert.*
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+@KotlinCleanup(" 'is'-> equalTo")
+@RunWith(AndroidJUnit4::class)
+class SchedTest : RobolectricTest() {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/sched/SchedTest.kt:66]: Add two more KotlinCleanup:
```
@KotlinCleanup("fix IDE lint issues")
@KotlinCleanup("improve kotlin code where possible")
```

**Review by lukstbit — CHANGES_REQUESTED** [review]

**Review by lukstbit — APPROVED** [review]
Ok!

**Comment by mikehardy** [comment]
This is an area of really fast change - which is both unfortunate (it keeps blowing up the PR, I'm so sorry) but also rare and amazing (as Damien has taken the time to make really big/important advances in the scheduler support with #10411 ) - it means this needs another scan though since I just merged #10411 so now scheduler code changed up again. I'm not aware of any other PRs against the scheduler at the moment, at least

**Comment by Arthur-Milchior** [comment]
And now there is a conflict, sorry

**Review by Arthur-Milchior — COMMENTED** [review]
Thanks a lot. A few nit if you please.

Your `assertEqual` longs, make me thought about https://github.com/ankidroid/Anki-Android/issues/11965

**Pull request closed without merging**